src/components/ListItem.jsx

```diff
@@ -1,26 +1,46 @@
-import { useEffect, useState } from 'react';
+import { useState, useEffect } from 'react';
+
 import { updateItem } from '../api/firebase';
+import { ONE_DAY_IN_MILLISECONDS } from '../utils';
 
 import './ListItem.css';
 
-export function ListItem({ name, listToken, itemId, dateLastPurchased }) {
-	const [wasPurchased, setWasPurchased] = useState(false);
+export function ListItem({
+	name,
+	listToken,
+	itemId,
+	dateLastPurchased,
+	dateNextPurchased,
+}) {
+	const [prevDateLastPurchased, setPrevDateLastPurchased] = useState(null);
+	const [prevDateNextPurchased, setPrevDateNextPurchased] = useState(null);
+	const [disabled, setDisabled] = useState(false);
 
-	const handleCheck = (checked) => {
-		updateItem(checked, listToken, itemId);
-		setWasPurchased((prevWasPurchased) => !prevWasPurchased);
-	};
+	const currentDate = new Date().getTime();
+	const dateLastPurchasedPlus24h = dateLastPurchased
+		? dateLastPurchased.toDate().getTime() + ONE_DAY_IN_MILLISECONDS
+		: null;
 
+	const wasPurchased = currentDate < dateLastPurchasedPlus24h;
+
+	//disables ability to uncheck item if the item was marked as purchased on page load
 	useEffect(() => {
-		const currentDate = new Date().getTime();
-		const dayInMilliSec = 60 * 60 * 24 * 1000;
+		setDisabled(wasPurchased);
+	}, []);
 
-		const dateLastPurchasedPlus24h = dateLastPurchased
-			? dateLastPurchased.toDate().getTime() + dayInMilliSec
-			: null;
+	const handleCheck = (checked) => {
```

Comment:
Another slightly superficial code style idea, since your handleCheck function is a closure around `wasPurchased`, you could forgo the `checked` argument and use `wasPurchased` directly. 

With that change, you could further clean up your event handle to `onChange={handleCheck}`